File: Groupe1/TP1/src/binaire.c
#include <stdio.h>

void afficher_binaire(int num) {
    // Si le nombre est négatif, on le convertit en positif pour l'affichage
    int number = (int)num; // Utilisation d'un unsigned pour éviter les problèmes avec les nombres négatifs
    char binaire[33]; // 32 bits + 1 pour le caractère nul
    int index = 0;

    // Remplir le tableau de bits en utilisant la méthode de division par 2
    for (int i = 0; number > 0; i++) {
        binaire[i] = (number % 2) + '0'; // Stocke le bit (0 ou 1) comme caractère
        number /= 2;
        index++;
    }


    // Affichage en ordre inverse
    printf("Représentation binaire de %d : ", number);
    for (int i = index - 1; i >= 0; i--) {
        printf("%c", binaire[i]);
    }
    printf("\n");
}

int main() {
    int nombre;

    // Saisie d'un nombre entier par l'utilisateur
    printf("Entrez un nombre entier : ");
    scanf("%d", &nombre);

    // Appel de la fonction pour afficher la représentation binaire
    afficher_binaire(nombre);

    return 0;
}

File: Groupe1/TP1/src/binaire.py
def afficher_binaire(n):
    # Si le nombre est négatif, on le convertit en positif pour l'affichage
    num = abs(n)
    binaire = ""

    # Utiliser une boucle for pour extraire les bits
    for i in range(num.bit_length()):  # bit_length() donne le nombre de bits nécessaires pour représenter le nombre
        binaire = str(num % 2) + binaire  # Ajoute le bit au début de la chaîne
        num //= 2  # Division entière par 2



    # Affichage de la représentation binaire
    print(f"Représentation binaire de {n} : {binaire}")

if __name__ == "__main__":
    # Saisie d'un nombre entier par l'utilisateur
    nombre = int(input("Entrez un nombre entier : "))

    # Appel de la fonction pour afficher la représentation binaire
    afficher_binaire(nombre)
